pydm/widgets/line_edit.py

```diff
@@ -162,14 +162,26 @@ def send_value(self):
 
     def setReadOnly(self, readOnly):
         self._user_set_read_only = readOnly
-        super().setReadOnly(True if self._user_set_read_only else not self._write_access)
+        shouldSetReadOnly = True if self._user_set_read_only else (not self._write_access)
+        # When in designer and reading in live data (DESIGNER_ONLINE), don't set and therefore save to
+        # the ui file readOnly=True setting. While we do want widgets to act in read-only way in designer
+        # (so don't accidentally write data during editing), this is handled in the data_plugins themselves.
+        if is_qt_designer() and config.DESIGNER_ONLINE:
+            shouldSetReadOnly = False
+        super().setReadOnly(shouldSetReadOnly)
```

Comment:
Maybe an indulgently nitpicky suggestion for reimplementing this in a python/pep8y way
```suggestion
        # When in designer and reading in live data (DESIGNER_ONLINE), don't set and therefore save to
        # the ui file readOnly=True setting. While we do want widgets to act in read-only way in designer
        # (so don't accidentally write data during editing), this is handled in the data_plugins themselves.
        if is_qt_designer() and config.DESIGNER_ONLINE:
            should_set_read_only = False
        else:
            should_set_read_only = self._user_set_read_only or not self._write_access
        super().setReadOnly(should_set_read_only)
```